src/code/models/link-manager.coffee

```diff
@@ -185,21 +185,28 @@ module.exports = class LinkManager
     for listener in @selectionListeners
       listener({node:@selectedNode, connection:null})
 
-  changeNode: (title, image) ->
+  changeNode: (data) ->
     if @selectedNode
       node = @selectedNode
-      originalTitle = @selectedNode.title
-      originalImage = @selectedNode.image
+      originalData =
+        title: node.title
+        image: node.image
+        color: node.color
```

Comment:
I wonder if it would be better to use _.cloneDeep when saving this reference so that any new data members automatically get copied.

    originalData = _.cloneDeep data